tests/testPtr.ml

```diff
@@ -0,0 +1,73 @@
+open OUnit2
+open TestLib
+
+let property_test_count = 100
+let qcheck_to_ounit t = OUnit.ounit2_of_ounit1 @@ QCheck_ounit.to_ounit_test t
+
+(* NOTE: Ideally we want the definition of ptr to be opaque. A sensible
+ * property to expect would be the following:
+ *
+ *   forall (n: nat) (m: nat).
+ *     n <> m =>
+ *     ptr_next (..n_times (ptr_next ptr_null)..) <> ptr_next (..m_times (ptr_next ptr_null)..)
```

Comment:
Hmmm, how would you write this formally?

Reply:
Do you mean that the function is 1-to-1? Extending the test above probably captures both directions, I'm thinking

Reply:
I guess I'm just thinking about whether or not `ptr_next` is required to never return the same output for a given initial ptr. I think that the current implementation has this property since it doesn't do any recycling of old pointers as far as I'm aware, though that might be more of a property of the implementation and not the `Ptr` interface :thinking:. If that's the case then what I said about a test for uniqueness wouldn't really be applicable

Reply:
Ah, I see what you mean. Yes, some of these properties (e.g. that `null` does not refer to any object) come from the particular implementation of `mem.ml` and how we (in theory) update it. We don't do any recycling, but we could accidentally do something like
```ocaml
let mem1 = ... mem0 ... in
let mem2 = ... mem1 ... in
(* and here we erroneously use mem1 instead of mem2 *)
```
which we'd like the tests to catch of course, but this is too far from `ptr.ml`.